crates/runtime/src/lib.rs

```diff
@@ -322,6 +324,13 @@ impl_runtime_apis! {
             };
             Ok(UncheckedExtrinsic::new_unsigned(call.into()))
         }
+
+        fn convert_error(error: DispatchError) -> StarknetRpcApiError {
+            if error == PalletError::<Runtime>::ContractNotFound.into() {
+                return StarknetRpcApiError::ContractNotFound;
+            }
+            StarknetRpcApiError::ContractError
+        }
```

Comment:
Is there a way to have other more descriptive errors? Not just for `ContractNotFound`